### PR TITLE
WFLY-5519 Some clustering tests fail with security manager

### PR DESCRIPTION
**Pull request by rhusar**
Jiras
https://issues.jboss.org/browse/WFLY-5519
https://issues.jboss.org/browse/WFLY-6025

### CODE REVIEW

**Comment by rhusar** [comment]
retest this please

**Comment by rhusar** [comment]
retest this please

**Comment by rhusar** [comment]
retest this please

**Comment by rhusar** [comment]
retest this please

**Comment by rhusar** [comment]
retest this please

**Comment by n1hility** [comment]
@pferraro Can you confirm this is good for 10.x?

File: clustering/server/src/main/java/org/wildfly/clustering/server/dispatcher/LocalCommandDispatcher.java [hunk 1]
```diff
@@ -114,6 +115,14 @@ public R call() throws Exception {
 
     @Override
     public void close() {
-        this.executor.shutdown();
+        PrivilegedAction<Void> closeAction = () -> {
+            this.executor.shutdown();
+            return null;
+        };
+        if (System.getSecurityManager() == null) {
```
Comment (pferraro) [inline clustering/server/src/main/java/org/wildfly/clustering/server/dispatcher/LocalCommandDispatcher.java]: We should leverage the appropriate org.wildfly.security.manager.WildFlySecurityManager method here.

**Comment by rhusar** [comment]
retest this please

File: clustering/server/src/main/java/org/wildfly/clustering/server/dispatcher/LocalCommandDispatcher.java [hunk 2]
```diff
@@ -114,6 +115,14 @@ public R call() throws Exception {
 
     @Override
     public void close() {
-        this.executor.shutdown();
+        PrivilegedAction<Void> closeAction = () -> {
+            this.executor.shutdown();
+            return null;
+        };
+        if (WildFlySecurityManager.isChecking()) {
```
Comment (pferraro) [inline clustering/server/src/main/java/org/wildfly/clustering/server/dispatcher/LocalCommandDispatcher.java]: Just use WildFlySecurityManager.doChecked(...), I think.

**Comment by pferraro** [comment]
:+1: This looks good now.

**Pull request merged**